### PR TITLE
Add support for documenting built-in annotations

### PR DESCRIPTION
**Pull request by YuriSizov**
I love annotations! <3 One thing that constantly annoys me in 3.x is that to look up export hint documentation you have to use the web docs. This should make things so much simpler. And later on, if (when?) we implement custom annotations, it should be trivial to add those to the docs as well.

https://user-images.githubusercontent.com/11782833/177191610-3f05672d-86c4-4114-9f8d-a9dcd8e1a6f0.mp4

Once this is merged, PRs for the actual descriptions are welcome! For starters, all `@export*` annotations can be copied over from the property exporting docs on web.

-----

There are a couple of unrelated changes aimed to bring a more consistency to the look of the docs. Descriptions are properly stripped before being checked for emptiness and bulletpoints have consistent colors now.

### CODE REVIEW

**Review by vnen — APPROVED** [review]
Looks good to me, at least for the GDScript changes

File: modules/gdscript/doc_classes/@GDScript.xml [hunk 1]
```diff
@@ -257,4 +257,145 @@
...
+			</description>
+		</annotation>
+		<annotation name="@export_range">
+			<return type="void" />
+			<argument index="0" name="min" type="float" />
+			<argument index="1" name="max" type="float" />
+			<argument index="2" name="step" type="float" default="null" />
+			<argument index="3" name="slider1" type="String" default="null" />
+			<argument index="4" name="slider2" type="String" default="null" />
+			<argument index="5" name="slider3" type="String" default="null" />
```
Comment (akien-mga) [inline modules/gdscript/doc_classes/@GDScript.xml:365]: Seems like you're not properly getting the default value information.

**Review by akien-mga — APPROVED** [review]
Looks great to me. I found a bug (see above comment) but this likely pre-exists this PR (depends on how annotations are registered with MethodInfo) and this can be tracked in another issue.

**Comment by KoBeWi** [comment]
Does this implement links, e.g. `[annotation @export]`? Doesn't seem to work.

**Comment by YuriSizov** [comment]
@KoBeWi Yes, I missed it here, but fixed it in the follow-up https://github.com/godotengine/godot/pull/64170

**Pull request merged**